### PR TITLE
Make `grakn.sh` wait for Grakn to completely initialise before returning

### PR DESCRIPTION
**Pull request by aelred**

### CODE REVIEW

File: grakn-dist/src/bin/grakn-engine.sh [hunk 1]
```diff
@@ -41,6 +41,26 @@ done
 # Add path containing logback.xml
 CLASSPATH="$CLASSPATH":"${GRAKN_HOME}"/conf/main/
 
+wait_for_engine() {
+    local now_s=`date '+%s'`
+    local stop_s=$(( $now_s + $ENGINE_STARTUP_TIMEOUT_S ))
+    local status_thrift=
```
Comment (burukuru) [inline grakn-dist/src/bin/grakn-engine.sh]: Don't need this.

**Comment by aelred** [comment]
!rtg - cassandra timeout

File: grakn-client/src/main/java/ai/grakn/client/Client.java [hunk 2]
```diff
@@ -55,7 +57,20 @@
             return reader.lines().collect(joining("\n"));
         }
     };
-    
+
+    public static void main(String[] args) throws IOException {
+        String confPath = System.getProperty("grakn.conf");
```
Comment (alexandraorth) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: Should we extract this property into core?

File: grakn-client/src/main/java/ai/grakn/client/Client.java [hunk 3]
```diff
@@ -55,7 +57,20 @@
...
+        Properties properties = new Properties();
+        try (FileInputStream stream = new FileInputStream(confPath)) {
```
Comment (alexandraorth) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: I think here we should check if the file exists and print a nice, helpful message if it does not.

**Comment by aelred** [comment]
@alexandraorth I tried to find some to remove, but only found one in the bash tests. The others (such as `DistributionContext`) do not call `grakn.sh`, so can't use our new fancy method.

File: grakn-client/src/main/java/ai/grakn/client/Client.java (hunk 2 above)
Comment (aelred) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: I _guess_ so...

File: grakn-client/src/main/java/ai/grakn/client/Client.java (hunk 3 above)
Comment (aelred) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: Good point!

File: grakn-client/src/main/java/ai/grakn/client/Client.java (hunk 2 above)
Comment (alexandraorth) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: What do you think @fppt ? Should we have configuration constants in core?

Comment (fppt) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: Put it somewhere yes. The less duplicated hardcoded strings we have the better. It's always a pain to change one and have to hunt down the 500 others you missed.

Comment (aelred) [inline grakn-client/src/main/java/ai/grakn/client/Client.java]: I've done it in a separate PR #1815. I'll wait for that to merge, then use it here (so don't merge this one yet!)

**Comment by aelred** [comment]
@alexandraorth @burukuru please re-review!

**Pull request merged**